Assets/Scripts/Player/PlayerControls.cs

```diff
@@ -403,16 +420,30 @@ public override void Exit()
 
 	void SetupStateMethodMap()
 	{
-		_actor.physics.RegisterState( PhysicsStateType.Jumping,  new Jumping( _actor ) );
-		_actor.physics.RegisterState( PhysicsStateType.Falling,  new Falling( _actor ) );
-		_actor.physics.RegisterState( PhysicsStateType.Grounded, new Grounded( _actor ) );
-		_actor.physics.RegisterState( PhysicsStateType.Climbing, new Climbing( _actor ) );
-		_actor.physics.RegisterState( PhysicsStateType.Gliding,  new Gliding( _actor ) );
-		_actor.physics.RegisterState( PhysicsStateType.Sprinting, new Sprinting( _actor ) );	
+		_actor.physics.RegisterState( PhysicsStateType.Jumping,   new Jumping( _actor ) );
+		_actor.physics.RegisterState( PhysicsStateType.Falling,   new Falling( _actor ) );
+		_actor.physics.RegisterState( PhysicsStateType.Grounded,  new Grounded( _actor ) );
+		_actor.physics.RegisterState( PhysicsStateType.Climbing,  new Climbing( _actor ) );
+		_actor.physics.RegisterState( PhysicsStateType.Gliding,   new Gliding( _actor ) );
+		_actor.physics.RegisterState( PhysicsStateType.Sprinting, new Sprinting( _actor ) );
 	}
 	#endregion
 
-	/**
+	public bool InteractCheck( RaycastHit hitInfo )
+	{
+		Interactable interactable = hitInfo.collider.gameObject.GetComponent<Interactable>();

```

Comment:
Can you rename this class to something more descriptive, like TextInteractable or TextPopup?